magpie/register.py

```diff
@@ -703,6 +705,10 @@ def _log_permission(message, permission_index, trail=", skipping...", detail=Non
     permission = " [{!s}]".format(permission) if permission else ""
     LOGGER.log(level, "%s [permission #%d]%s%s", message, permission_index, permission, trail)
 
+    if raise_errors:
+        raise RegistrationConfigurationError(
+            "{} [permission #{}]{}{}".format(message, permission_index, permission, trail))
```

Comment:
You can use the `msg = ...` with this format and pass it to both log/exception. 
No point to "optimize" the log call in this case.